src/app/tests/suites/certification/Test_TC_SMCO_2_2.yaml

```diff
@@ -0,0 +1,286 @@
+      command: "readAttribute"
+      attribute: "ExpressedState"
+      response:
+          value: 1
+          constraints:
+              type: enum8
+
+    - label: "TH reads SmokeAlarm event from DUT"
+      PICS: SMOKECO.S.E00
+      endpoint: 1
+      command: "readEvent"
+      event: "SmokeAlarm"
+      eventNumber: eventNumber + 1
+      response:
+          value: { AlarmSeverityLevel: 1 }
+
+    - label: "Start manually DUT self-test"
+      cluster: "LogCommands"
+      PICS: PICS_USER_PROMPT && SMCO.M.ManuallyControlledTest
+      endpoint: 1
+      command: "UserPrompt"
+      arguments:
+          values:
+              - name: "message"
+                value: "Please enter 'y' for success"
+              - name: "expectedValue"
+                value: "y"
+
+    - label: "TH reads TestInProgress attribute from DUT"
+      PICS: SMOKECO.S.A0005
+      endpoint: 1
+      command: "readAttribute"
+      attribute: "TestInProgress"
+      response:
+          value: 0
+          constraints:
+              type: boolean
+
+    - label: "TH sends SelfTestRequest command to DUT"
+      PICS: SMOKECO.S.C00.Rsp
+      endpoint: 1
+      command: "SelfTestRequest"
+      response:
+          error: BUSY
+
+    - label: "TH reads TestInProgress attribute from DUT"
+      PICS: SMOKECO.S.A0005
+      endpoint: 1
+      command: "readAttribute"
+      attribute: "TestInProgress"
+      response:
+          value: 0
+          constraints:
+              type: boolean
+
+    - label:
+          "TH sends TestEventTrigger command to General Diagnostics Cluster on
+          Endpoint 0 with EnableKey field set to
+          PIXIT.SMCO.TEST_EVENT_TRIGGER_KEY and EventTrigger field set to
+          PIXIT.SMCO.TEST_EVENT_TRIGGER for Critical Smoke Alarm Test Event"
+      PICS: DGGEN.S.C00.Rsp
+      cluster: "General Diagnostics"
+      endpoint: 0
+      command: "TestEventTrigger"
+      arguments:
+          values:
+              - name: "EnableKey"
+                value: TEST_EVENT_TRIGGER_KEY
+              - name: "EventTrigger"
+                value: TEST_EVENT_TRIGGER_CRITICAL_SMOKE_ALARM
+
+    - label:
+          "TH waits for a report of SmokeState attribute from DUT with a timeout
+          of 300 seconds"
+      PICS: SMOKECO.S.A0001
+      endpoint: 1
+      command: "waitForReport"
+      attribute: "SmokeState"
+      timeout: 300
+      response:
+          value: 2
+          constraints:
+              type: enum8
+
+    - label: "TH reads ExpressedState attribute from DUT"
+      PICS: SMOKECO.S.A0000
+      endpoint: 1
+      command: "readAttribute"
+      attribute: "ExpressedState"
+      response:
+          value: 1
+          constraints:
+              type: enum8
+
+    - label: "TH reads SmokeAlarm event from DUT"
+      PICS: SMOKECO.S.E00
+      endpoint: 1
+      command: "readEvent"
+      event: "SmokeAlarm"
+      eventNumber: eventNumber + 2
+      response:
+          value: { AlarmSeverityLevel: 2 }
```

Comment:
readEvent will read _all events from the device_, no?  Or is the `eventNumber` bit there doing filtering on the list somehow?  @vivien-apple

Reply:
The purpose of getting the last eventNumber is to use it to filter events and ensure that it passes the test multiple times.

Reply:
No my point was that my understanding was that the "response" to a readEvent was a list of events.  If you specify eventNumber there, does it change to having only a single event as a response instead?

Reply:
Yes.